src/emu/tilemap.h

```diff
@@ -444,11 +448,12 @@ struct tile_data
 	UINT8           flags;          // defaults to 0; one or more of TILE_* flags above
 	UINT8           pen_mask;       // defaults to 0xff; mask to apply to pen_data while rendering the tile
 	UINT8           gfxnum;         // defaults to 0xff; specify index of gfx for auto-invalidation on dirty
+	UINT32          code;
 
 	void set(int _gfxnum, int rawcode, int rawcolor, int _flags)
 	{
 		gfx_element *gfx = decoder->gfx(_gfxnum);
-		int code = rawcode % gfx->elements();
+		code = rawcode % gfx->elements();
```

Comment:
This is playing fast and loose with integer types, and making an assumption about how the code is mapped by the gfx decoder.  The whole thing is being dodgy with regards to integers - casting straight from int to UINT8 is undefined for negative values.